### DISCUSSION TITLE
IPAM / Services & Service Templates - Allow multiple protocols per service

### DISCUSSION THREAD

**Discussion by mjf** [netbox-community/netbox#17380] [Ideas]
It would be handy if we could use multiple protocols for IPAM / Services (_e.g._, both TCP and UDP for DNS port 53). Is it possible to extend the IPAM / Service model to allow for multiple protocols?

-----

**Off-topic sidenote**: I also feel like sort of discrepancy exists between NetBox Service which in fact just documents a port on an interface where some "service" (or daemon providing some "service", "function" or "protocol") is listening at, and that documents that fact only and can only be used as such. But for me a service is something quite different and typically composes from several components and forms the real service (_e.g._, some service may expose REST API at HTTPS together with many other ports with many other protocols to communicate with it) and some of the components may be "hidden" (_e.g._, some API backend using database through local unix socket would also be part of the service). That is something we can not model in NetBox right now (except using some plugin) and the current IPAM / Service and Service Template naming may collide with some realistic service model that may be (as I hope) added in the future. Therefor I also propose to rename the IPAM / Service to something else that would reflect more the nature of this model.

**Comment by ghost** [comment]
way ahead of you : https://github.com/netbox-community/netbox/issues/17196

I originally just submitted a FR for your exact use case here (https://github.com/netbox-community/netbox/issues/16841) but the above FR expands the scope and makes services pretty darn useful.
